src/components/forms/NumberField/NumberField.js

```diff
@@ -0,0 +1,99 @@
+import {
+  FormFieldDescription,
+  FormLabel,
+  Paragraph,
+  Textbox,
+  FormField as UtrechtFormField,
+} from '@utrecht/component-library-react';
+import {useField} from 'formik';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {useIntl} from 'react-intl';
+import {NumericFormat} from 'react-number-format';
+
+import {getBEMClassName} from 'utils';
+
+const NumberField = ({
+  name,
+  min,
+  step,
+  label = '',
+  isRequired = false,
+  description = '',
+  id = '',
+  disabled = false,
+  invalid = false,
+}) => {
+  const [field, meta, helpers] = useField(name);
+  const {locale} = useIntl();
+
+  const numberFormat = new Intl.NumberFormat(locale);
+  const labelClassName = getBEMClassName('label', [isRequired && 'required'].filter(Boolean));
+  const inputClassName = getBEMClassName('input', [invalid && 'invalid'].filter(Boolean));
+
+  const allowNegative = min && min < 0;
+  // We get the decimal separator by formatting a arbitrary number, and then extracting the decimal separator
+  const decimalSeparator = numberFormat
+    .formatToParts(1.1)
+    .find(part => part.type === 'decimal').value;
+  const thousandSeparator = decimalSeparator === ',' ? '.' : ',';
+  const isAllowedToTypeDecimals = !step;
+
+  // This function makes sure that the user can't type a number that is lower than the min value
+  const isAllowed = values => {
+    const {formattedValue, floatValue} = values;
+    const isEmpty = formattedValue === '';
+    const isBelowMin = min ? floatValue >= min : true;
```

Comment:
should check if `min` is a number instead of truthy, because with `min=0` you end up in the `true` result instead of checking if the value is greater than or equal than 0